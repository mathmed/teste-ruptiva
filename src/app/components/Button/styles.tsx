import styled from 'styled-components/native';

export const Container = styled.TouchableOpacity`
  background: #ff9000;
  border-radius: 10px;
  height: 56px;
  padding: 0 16px;
  width: 100%;
  margin-top: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
