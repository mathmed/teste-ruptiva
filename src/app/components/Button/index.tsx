import React from 'react';
import { TouchableOpacityProps } from 'react-native';
import { Container } from './styles';

type CustomButtonProps = TouchableOpacityProps;

const Button: React.FC<CustomButtonProps> = ({ children, ...props }) => (
  <Container {...props}>{children}</Container>
);

export default Button;
