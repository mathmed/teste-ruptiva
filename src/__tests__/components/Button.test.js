import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../app/components/Button';

let screen = null;
jest.useFakeTimers();
describe('[COMPONENTS] Button', () => {
  beforeEach(async () => {
    screen = await renderer.create(<Button />);
  });
  it('Should render corretly', async () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
