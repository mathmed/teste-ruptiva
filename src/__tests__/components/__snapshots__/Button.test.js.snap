// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] Button Should render corretly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ff9000",
      "borderRadius": 10,
      "display": "flex",
      "height": 56,
      "justifyContent": "center",
      "marginTop": 10,
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 0,
      "width": "100%",
    }
  }
/>
`;
